omou-frontend/src/components/FeatureViews/Scheduler/Scheduler.js

```diff
@@ -291,123 +292,119 @@ class Scheduler extends Component {
+                    <br />
+                    <Grid container
+                        direction="row"
+                        alignItems="center"
+                        className="scheduler-header">
+                        <Grid item className={"calendar-icon-header"}>
+                            <IconButton color={this.state.calendarIcon ? "primary" : ""} onClick={this.changeViewToCalendar} className={'calendar-icon'} aria-label='next-month'>
+                                <DateRangeOutlinedIcon />
+                            </IconButton>
+                        </Grid>
+                        <Grid item  >
+                            <IconButton color={this.state.resourceIcon ? "primary" : ""} onClick={this.changeViewToResource} className={'resource-icon'} aria-label='next-month'>
+                                <ViewListIcon />
+                            </IconButton>
+                        </Grid>
+                        <Grid item  >
+                            <IconButton onClick={""} className={'next-month'} aria-label='next-month'>
+                                <SearchIcon />
+                            </IconButton>
+                        </Grid>
+                        <Grid item sm={1}>
+                            {(this.state.calendarIcon) ?
+                                <FormControl className={'filter-select'}>
+                                    <InputLabel htmlFor="filter-class-type"></InputLabel>
 
-                            </Grid>
-                            <Grid item lg={1} md={1}>
-                                <IconButton onClick={this.goToPrev} className={'prev-month'} aria-label="prev-month">
-                                    <ChevronLeftOutlinedIcon />
-                                </IconButton>
-                            </Grid>
-                            <Grid item lg={3} md={2}>
-                                <Typography variant={'h6'}>  {this.state.currentDate} </Typography>
-                            </Grid>
-                            <Grid item lg={1} md={1} >
-                                <IconButton onClick={this.goToNext} className={'next-month'} aria-label='next-month'>
-                                    <ChevronRightOutlinedIcon />
-                                </IconButton>
-                            </Grid>
-                            <Grid item md={2} lg={1}>
-                                <FormControl className={'change-view'}>
-                                    <InputLabel htmlFor="change-view-select"></InputLabel>
                                     <Select
                                         native
-                                        value={this.state.viewValue}
-                                        onChange={(event) => this.changeView(event.target.value)}
+                                        value={this.state.filterValue}
+                                        onChange={this.handleFilterChange('filterValue')}
```

Comment:
```
handleFilterChange = (name) => event => {
    console.log(event)
    this.setState({
        ...this.state,
        [name]: event.target.value
    })
    this.filterEvent(event.target.value)
}
```
Regarding this function, as it is only used here, remove the function-that-returns-a-function complexity and simply do
```
this.setState({
    "filterValue": event.target.value
})
```
Also, setState merges the given object with the current state, so no need for the `...this.state`. Only the specified properties are updated.

Note that you will have to change this to `onChange={this.handleFilterChange}` to ensure that this is still a function.